shared/chat/conversation/input-area/normal/container.js

```diff
@@ -117,24 +77,15 @@ const mapDispatchToProps = (dispatch: Dispatch) => ({
 const mergeProps = (stateProps, dispatchProps, ownProps: OwnProps): Props => ({
   conversationIDKey: stateProps.conversationIDKey,
   channelName: stateProps._meta.channelname,
-  isEditing: !!stateProps._editingMessage,
+  isEditing: !!stateProps._editOrdinal,
   focusInputCounter: ownProps.focusInputCounter,
   clearInboxFilter: dispatchProps.clearInboxFilter,
   onAttach: (paths: Array<string>) => dispatchProps._onAttach(stateProps.conversationIDKey, paths),
   onEditLastMessage: () => dispatchProps._onEditLastMessage(stateProps.conversationIDKey, stateProps._you),
-  onCancelEditing: () => {
-    dispatchProps._onCancelQuoting(stateProps.conversationIDKey)
-    dispatchProps._onCancelEditing(stateProps.conversationIDKey)
-  },
-  onCancelQuoting: () => dispatchProps._onCancelQuoting(stateProps.conversationIDKey),
+  onCancelEditing: () => dispatchProps._onCancelEditing(stateProps.conversationIDKey),
   onSubmit: (text: string) => {
-    const em = stateProps._editingMessage
-    if (em) {
-      if (em.type === 'text' && em.text.stringValue() === text) {
```

Comment:
onEditMessage already does this check

Reply:
where's that?

Reply:
I mean `dispatchProps._onEditMessage`, which calls `messageEdit` in `actions/chat2/index.js`.